src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -1572,6 +1573,10 @@ internal void Validate()
                 throw new ArgumentException("Host can't be null");
             if (Multiplexing && !Pooling)
                 throw new ArgumentException("Pooling must be on to use multiplexing");
+            if (SslMode == SslMode.Require && !TrustServerCertificate.HasValue)
+            {
+                throw new NpgsqlException("SslMode.Require requires TrustServerCertificate to be explicitly set. Please see https://www.npgsql.org/doc/release-notes/6.0.html");
```

Comment:
```suggestion
                throw new NpgsqlException(
                    "To validate server certificates, please use VerifyFull or VerifyCA instead of Require. " +
                    "To disable validation, explicitly set 'Trust Server Certificate' to true. " +
                    "See https://www.npgsql.org/doc/release-notes/6.0.html for more details.");
```